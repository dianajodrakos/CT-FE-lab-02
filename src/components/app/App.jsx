import React from 'react';
import AnimalCrossingContainer from '../../containers/AnimalCrossing';

export default function App() {
  return <AnimalCrossingContainer />;
}
