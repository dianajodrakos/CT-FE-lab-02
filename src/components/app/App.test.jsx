/* eslint-disable max-len */
import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { act } from 'react-dom/test-utils';
import { MemoryRouter } from 'react-router';
import listData from '../../fixtures/listData.json';
import detailData from '../../fixtures/detailData.json';
import App from './App';

const server = setupServer(
  rest.get('https://ac-vill.herokuapp.com/villagers/', (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json(listData)
    );
  }),
  rest.get('https://ac-vill.herokuapp.com/villagers/5f5fb4bbbfd05c2aed82e462', (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json(detailData)
    );
  })
);

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Animal Crossing App', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('renders a list of villagers', async () => {
    act(() => {
      render(
        <MemoryRouter>
          <App />
        </MemoryRouter>
      );
    });

    screen.getByAltText('loading');

    const ul = await screen.findByRole(
      'list',
      { name: 'villagers' },
    );

    expect(ul).toMatchSnapshot();
  });

  it('renders a single villager', async () => {
    act(() => {
      render(
        <MemoryRouter initialEntries={['/Agnes']}>
          <App />
        </MemoryRouter>
      );
    });

    screen.getByAltText('loading');

    const h2 = await screen.findByRole('heading');

    expect(h2).toMatchSnapshot();
  });
});
