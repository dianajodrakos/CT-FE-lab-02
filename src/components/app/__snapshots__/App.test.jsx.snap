// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal Crossing App renders a list of villagers with pagination 1`] = `
<ul
  aria-label="villagers"
/>
`;

exports[`Animal Crossing App renders a single villager 1`] = `
<h1>
  Animal Crossing Villagers
</h1>
`;

exports[`App component renders App 1`] = `
<DocumentFragment>
  <div>
    <a
      href="/"
    >
      <h1>
        Animal Crossing Villagers
      </h1>
    </a>
  </div>
  <div>
    <img
      alt="loading"
      height="250px"
      src="https://furrycdn.org/img/view/2020/11/19/48649.gif"
    />
  </div>
</DocumentFragment>
`;
